frontend/src/views/DashboardPage/DashboardPage.tsx

```diff
@@ -961,28 +1205,32 @@ export const DashboardPage = () => {
                       folders={folderList}
                       search={searchFilter}
                     />
-                    {fields.map(({ id, _id }, index) => (
-                      <SecretInputRow
-                        key={id}
-                        secUniqId={_id}
-                        isReadOnly={isReadOnly}
-                        isRollbackMode={isRollbackMode}
-                        isAddOnly={isAddOnly}
-                        index={index}
-                        searchTerm={searchFilter}
-                        onSecretDelete={onSecretDelete}
-                        isKeyError={Boolean(errors?.secrets?.[index]?.key?.message)}
-                        keyError={errors?.secrets?.[index]?.key?.message}
-                        onRowExpand={onDrawerOpen}
-                        isSecretValueHidden={isSecretValueHidden}
-                        wsTags={wsTags}
-                        onCreateTagOpen={handleCreateTagModalOpen}
-                        register={register}
-                        control={control}
-                        setValue={setValue}
-                        autoCapitalization={currentWorkspace?.autoCapitalization}
-                      />
-                    ))}
+                    {fields.map(({ id, _id }, index) => {
+                      return (
```

Comment:
Can you check the rendering. I feel this will cause each input to render back to back

Reply:
Yeah, I think this looks okay. This is how the fields have been rendered before now.

Reply:
Two things here

1. The return can u make it back
2. Kindly use the react developer tool and here the new props you have passed are causing re-rendering of all inputs whenever a Append or Delete of input happens which will slow down the dashboard speed. Kindly optimize the props passed from inline fns